include/bout/field3d.hxx

```diff
@@ -514,6 +527,13 @@ private:
 
   /// RegionID over which the field is valid
   std::optional<size_t> regionID;
+
+  int tracking_state{0};
+  Options* tracking{nullptr};
```

Comment:
I really don't like storing raw pointers. Probably this makes most sense as `weak_ptr`, a non-owning reference to a `shared_ptr` allocated and owned elsewhere. That way it's very obvious we don't own it and we can rely on it living long enough